src/Libraries/Microsoft.Extensions.TimeProvider.Testing/README.md

```diff
@@ -33,31 +33,109 @@ public void Advance(TimeSpan delta)
 public void SetLocalTimeZone(TimeZoneInfo localTimeZone)
 ```
 
-These can be used as follows:
+### ExpiryCache with TimeProvider
+
+The example below demonstrates the `ExpiryCache` class and how it can be tested using `FakeTimeProvider` in `ExpiryCacheTests`. 
+
+The `TimeProvider` abstraction is injected into the `ExpiryCache` class, allowing the cache to rely on `GetUtcNow()` to determine whether cache entries should be evicted based on the current time. This abstraction provides flexibility by enabling different time-related behaviors in test environments.
+
+By using `FakeTimeProvider` in testing, we can simulate the passage of time with methods like `Advance()` and `SetUtcNow()`. This makes it possible to emulate the system's time in a controlled and predictable way during tests, ensuring that cache eviction works as expected.
 
 ```csharp
-var timeProvider = new FakeTimeProvider();
-var myComponent = new MyComponent(timeProvider);
-timeProvider.Advance(TimeSpan.FromSeconds(5));
-myComponent.CheckState();
+public class ExpiryCache<TKey, TValue>
+{
+    private readonly TimeProvider _timeProvider;
+    private readonly ConcurrentDictionary<TKey, CacheItem> _cache = new();
+    private readonly TimeSpan _expirationDuration;
+
+    public ExpiryCache(TimeProvider timeProvider, TimeSpan expirationDuration)
+    {
+        _timeProvider = timeProvider ?? throw new ArgumentNullException(nameof(timeProvider));
+        _expirationDuration = expirationDuration;
+    }
+
+    public void Add(TKey key, TValue value)
+    {
+        var expirationTime = _timeProvider.GetUtcNow() + _expirationDuration;
+        var cacheItem = new CacheItem(value, expirationTime);
+
+        _cache[key] = cacheItem;
+    }
+
+    public bool TryGetValue(TKey key, out TValue value)
+    {
+        value = default;
+        if (_cache.TryGetValue(key, out var cacheItem))
```

Comment:
We don't use `var` unless the type is obvious.
Please refer to your [style guide](https://github.com/dotnet/extensions/blob/main/CONTRIBUTING.md#dos-and-donts).

Reply:
Fixed